### PR TITLE
feat(static): add arc20token and implementation

### PR DESCRIPTION
**Pull request by JoaquinTissera-Simtlix**
add arc20token and implementation

STLX-9140

### CODE REVIEW

File: modules/statics/src/account.ts [hunk 1]
```diff
@@ -254,6 +258,22 @@ export class SolCoin extends AccountCoinToken {
...
+export class ARC20Token extends ContractAddressDefinedToken {
+  public assetIdXChain: string;
+  constructor(options: ARC20CoinContructorOptions) {
+    super({
+      ...options,
+    });
+
+    this.assetIdXChain = options.assetIdXChain;
+  }
+}
```
Comment (jrafael-simtlix) [inline modules/statics/src/account.ts:275]: Should we support regular ERC20 token in avax?
Should ARC20 token extends ERC20 ?
if both answers are yes, i suggest this change.

```suggestion
/**
 * The AVAX C Chain network support tokens
 * AVAX C Chain Tokens are ERC20 coin
 */
export class AvaxERC20Token extends ContractAddressDefinedToken {
  constructor(options: ERC20CoinContructorOptions) {
    super({
      ...options,
    });
}
}

/**
 * The AVAX C chain network support tokens
 * AVAX Native Tokens work similar to ERC20 coin, but need
 * to asset dXChain to the CrossChain
 */
export class ARC20Token extends AvaxERC20Token {
  public assetIdXChain: string;
  constructor(options: ARC20CoinContructorOptions) {
    super({
      ...options,
    });

    this.assetIdXChain = options.assetIdXChain;
  }
}
```

File: modules/statics/src/coins.ts [hunk 2]
```diff
@@ -1117,4 +1119,24 @@ export const coins = CoinMap.fromCoins([
   teosToken('teos:IQ', 'Everipedia', 3, 'testtoken112', UnderlyingAsset.IQ, AccountCoin.DEFAULT_FEATURES, '', 'IQ'),
   teosToken('teos:BOX', 'Box', 6, 'kvszn1xyz1bu', UnderlyingAsset.EOS_BOX, AccountCoin.DEFAULT_FEATURES, '', 'BOX'),
   teosToken('teos:USDT', 'Tether', 4, 'lionteste212', UnderlyingAsset.USDT, AccountCoin.DEFAULT_FEATURES, '', 'USDT'),
+  tavaxErc20('Qi', 'BenQi', 18, '0x8729438eb15e2c8b576fcc6aecda6a148776c0f5', UnderlyingAsset.QI),
+  tavaxErc20('XAVA', 'Avalaunch', 18, '0xd1c3f94de7e5b45fa4edbba472491a9f4b166fc4', UnderlyingAsset.XAVA),
+  tavaxErc20('KLO', 'Kalao', 18, '0xb27c8941a7df8958a1778c0259f76d1f8b711c35', UnderlyingAsset.KLO),
+  tavaxErc20('JOE', 'Trader Joe', 18, '0x6e84a6216ea6dacc71ee8e6b0a5b7322eebc0fdd', UnderlyingAsset.JOE),
```
Comment (jrafael-simtlix) [inline modules/statics/src/coins.ts:1125]: I see that same address works for testnet and mainnet.
Should we add mainnet token also?
https://avascan.info/blockchain/c/address/0x6e84a6216ea6dacc71ee8e6b0a5b7322eebc0fdd
https://testnet.avascan.info/blockchain/c/address/0x6e84a6216ea6dacc71ee8e6b0a5b7322eebc0fdd

File: modules/statics/src/account.ts [hunk 3]
```diff
@@ -888,3 +920,162 @@ export function tsolToken(
...
+export function tavaxErc20(
+  name: string,
+  fullName: string,
+  decimalPlaces: number,
+  contractAddress: string,
+  asset: UnderlyingAsset,
+  features: CoinFeature[] = AccountCoin.DEFAULT_FEATURES,
+  prefix: string = '',
+  suffix: string = name.toUpperCase(),
+  network: AccountNetwork = Networks.test.avalancheC
```
Comment (rliberal-simtlix) [inline modules/statics/src/account.ts:989]: keycurve here is not added. Why?

File: modules/statics/src/account.ts [hunk 4]
```diff
@@ -888,3 +920,162 @@ export function tsolToken(
...
+ *
+ * @param name unique identifier of the token
+ * @param fullName Complete human-readable name of the token
+ * @param decimalPlaces Number of decimal places this token supports (divisibility exponent)
+ * @param contractAddress Contract address of this token
+ * @param asset Asset which this coin represents. This is the same for both mainnet and testnet variants of a coin.
+ * @param prefix? Optional token prefix. Defaults to empty string
+ * @param suffix? Optional token suffix. Defaults to token name.
+ * @param network? Optional token network. Defaults to the Kovan test network.
+ * @param features? Features of this coin. Defaults to the DEFAULT_FEATURES defined in `AccountCoin`
```
Comment (rliberal-simtlix) [inline modules/statics/src/account.ts:978]: Again here keycurve

File: modules/statics/src/account.ts [hunk 5]
```diff
@@ -888,3 +920,162 @@ export function tsolToken(
...
+  fullName: string,
+  decimalPlaces: number,
+  contractAddress: string,
+  asset: UnderlyingAsset,
+  features: CoinFeature[] = AccountCoin.DEFAULT_FEATURES,
+  prefix: string = '',
+  suffix: string = name.toUpperCase(),
+  network: AccountNetwork = Networks.test.avalancheC
+) {
+  return avaxErc20(name, fullName, decimalPlaces, contractAddress, asset, features, prefix, suffix, network);
```
Comment (rliberal-simtlix) [inline modules/statics/src/account.ts:991]: keyCurve?

File: modules/statics/src/coins.ts [hunk 6]
```diff
@@ -1117,4 +1119,24 @@ export const coins = CoinMap.fromCoins([
   teosToken('teos:IQ', 'Everipedia', 3, 'testtoken112', UnderlyingAsset.IQ, AccountCoin.DEFAULT_FEATURES, '', 'IQ'),
   teosToken('teos:BOX', 'Box', 6, 'kvszn1xyz1bu', UnderlyingAsset.EOS_BOX, AccountCoin.DEFAULT_FEATURES, '', 'BOX'),
   teosToken('teos:USDT', 'Tether', 4, 'lionteste212', UnderlyingAsset.USDT, AccountCoin.DEFAULT_FEATURES, '', 'USDT'),
+  tavaxErc20('Qi', 'BenQi', 18, '0x8729438eb15e2c8b576fcc6aecda6a148776c0f5', UnderlyingAsset.QI),
```
Comment (charly-simtlix) [inline modules/statics/src/coins.ts:1122]: Should the 'i' be in uppercase?

File: modules/statics/src/coins.ts [hunk 7]
```diff
@@ -1117,4 +1119,24 @@ export const coins = CoinMap.fromCoins([
...
+  tarc20Token(
+    'PNG',
+    'Pangolin',
+    18,
+    '0x60781c2586d68229fde47564546784ab3faca982',
+    '2ugMhQ5ygoZ721Hjka413YkzNVGMjKQzmaDkxpeimKYEnAaWiD',
+    UnderlyingAsset.PNG
+  ),
+  tarc20Token(
+    'USDT',
```
Comment (charly-simtlix) [inline modules/statics/src/coins.ts:1135]: Is USDC missing?

File: modules/statics/src/coins.ts (hunk 2 above)
Comment (charly-simtlix) [reply modules/statics/src/coins.ts:1125]: Does the definition on mainnet have any impact on production now?
My concern if it is a way to enable/disable tokens in production until they are fully supported (all the development done)

File: modules/statics/test/unit/coins.ts [hunk 8]
```diff
@@ -61,4 +61,78 @@ describe('All Coins', () => {
     dot.decimalPlaces.should.equal(12);
     dot.primaryKeyCurve.should.equal('ed25519');
   });
+
+  describe('ARC20 and AVAXERC20 Tokens', () => {
+    it('Token PNG', () => {
```
Comment (charly-simtlix) [inline modules/statics/test/unit/coins.ts:66]: I recommend the description in the "it" block to be a verb in imperative tense:

`it('should contain definition for Token PNG')`

That way the test case is equivalent to a Business Rule that the software should obey.

Same comment for following tests

File: modules/statics/test/unit/coins.ts [hunk 9]
```diff
@@ -61,4 +61,78 @@ describe('All Coins', () => {
...
+      png.isToken.should.true;
+      png.asset.should.equal('png');
+      png.decimalPlaces.should.equal(18);
+      png.network.name.should.equal('AvalancheCTestnet');
+      png.network.type.should.equal('testnet');
+      png.network.family.should.equal('avaxc');
+    });
+
+    it('Token Qi', () => {
+      const qi = coins.get('Qi');
```
Comment (charly-simtlix) [inline modules/statics/test/unit/coins.ts:79]: Should the 'i' be in uppercase in the token name?

File: modules/statics/src/account.ts [hunk 10]
```diff
@@ -888,3 +920,162 @@ export function tsolToken(
...
+ * Factory function for avaxErc20 token instances.
+ *
+ * @param name unique identifier of the token
+ * @param fullName Complete human-readable name of the token
+ * @param decimalPlaces Number of decimal places this token supports (divisibility exponent)
+ * @param contractAddress Contract address of this token
+ * @param asset Asset which this coin represents. This is the same for both mainnet and testnet variants of a coin.
+ * @param prefix? Optional token prefix. Defaults to empty string
+ * @param suffix? Optional token suffix. Defaults to token name.
+ * @param network? Optional token network. Defaults to Ethereum main network.
```
[the remark this answers is not in the document]
Comment (rliberal-simtlix) [reply modules/statics/src/account.ts:934]: Related to network. Should be ETH network?

File: modules/statics/src/account.ts [hunk 11]
```diff
@@ -254,6 +258,34 @@ export class SolCoin extends AccountCoinToken {
   }
 }
 
+/**
+ * The AVAX C Chain network support tokens
+ * AVAX C Chain Tokens are ERC20 coin
+ */
+export class AvaxERC20Token extends ContractAddressDefinedToken {
+  constructor(options: Erc20ConstructorOptions) {
+    super({
```
Comment (charly-simtlix) [inline modules/statics/src/account.ts:267]: Idem

File: modules/statics/src/coins.ts [hunk 12]
```diff
@@ -1117,4 +1121,44 @@ export const coins = CoinMap.fromCoins([
...
+  ),
+  arc20Token(
+    'USDT',
+    'Tether',
+    6,
+    '0xc7198437980c041c805a1edcba50c1ce5db95118',
+    '2CWanxxHzCuagcyNCqpvQP4GWrxsikGua8hv6r3dYML5DegXsm',
+    UnderlyingAsset.USDT
+  ),
+  tavaxErc20('tQI', 'BenQi', 18, '0x8729438eb15e2c8b576fcc6aecda6a148776c0f5', UnderlyingAsset.QI),
```
Comment (charly-simtlix) [inline modules/statics/src/coins.ts:1144]: In the description of tokens for testnet is usual the adding of "Test" in the fullName to distinguish them from main (production) ones:
` terc20('tcat', 'Test CAT-20 Token', 18, '0x63137319f3a14a985eb31547370e0e3bd39b03b8', UnderlyingAsset.CAT)`

File: modules/statics/src/account.ts [hunk 13]
```diff
@@ -254,6 +258,33 @@ export class SolCoin extends AccountCoinToken {
...
+      ...options,
+    });
+  }
+}
+
+/**
+ * The AVAX C chain network support tokens
+ * AVAX Native Tokens work similar to ERC20 coin, but need
+ * to asset dXChain to the CrossChain
```
[the remark this answers is not in the document]
Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/account.ts]: Done

File: modules/statics/src/account.ts [hunk 14]
```diff
@@ -254,6 +258,34 @@ export class SolCoin extends AccountCoinToken {
   }
 }
 
+/**
+ * The AVAX C Chain network support tokens
+ * AVAX C Chain Tokens are ERC20 coin
```
[the remark this answers is not in the document]
Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/account.ts]: Done

File: modules/statics/src/account.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/account.ts]: Modify the network

File: modules/statics/src/coins.ts (hunk 6 above)
Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/coins.ts]: Done

File: modules/statics/test/unit/coins.ts (hunk 8 above)
Comment (JoaquinTissera-Simtlix) [reply modules/statics/test/unit/coins.ts]: Done

File: modules/statics/test/unit/coins.ts (hunk 9 above)
Comment (JoaquinTissera-Simtlix) [reply modules/statics/test/unit/coins.ts]: Done

File: modules/statics/src/account.ts (hunk 11 above)
Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/account.ts]: Done

File: modules/statics/src/coins.ts (hunk 12 above)
Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/coins.ts]: Done

File: modules/statics/test/unit/coins.ts [hunk 15]
```diff
@@ -35,3 +35,154 @@ describe('ERC20 Coins', () => {
       }, {});
   });
 });
+
+describe('All Coins', () => {
```
Comment (tylerlevine) [inline modules/statics/test/unit/coins.ts:39]: Remove all these tests, they are unnecessary

**Review by tylerlevine — CHANGES_REQUESTED** [review]

File: modules/statics/test/unit/coins.ts (hunk 15 above)
Comment (JoaquinTissera-Simtlix) [reply modules/statics/test/unit/coins.ts]: Done

**Review by shiksb — CHANGES_REQUESTED** [review]

File: modules/statics/src/base.ts [hunk 16]
```diff
@@ -420,11 +420,13 @@ export enum UnderlyingAsset {
   IVY = 'ivy',
   JBC = 'jbc',
   JFIN = 'jfin',
+  JOE = 'joe',
```
Comment (shiksb) [inline modules/statics/src/base.ts:423]: why are we adding these to ERC-20? Please add them add at the bottom (look at 703-726 as example). Add them like avaxc-bridge:JOE

**Review by shiksb — CHANGES_REQUESTED** [review]

File: modules/statics/src/coins.ts [hunk 17]
```diff
@@ -1117,4 +1121,44 @@ export const coins = CoinMap.fromCoins([
...
+    UnderlyingAsset.PNG
+  ),
+  arc20Token(
+    'USDT',
+    'Tether',
+    6,
+    '0xc7198437980c041c805a1edcba50c1ce5db95118',
+    '2CWanxxHzCuagcyNCqpvQP4GWrxsikGua8hv6r3dYML5DegXsm',
+    UnderlyingAsset.USDT
+  ),
```
[the remark this answers is not in the document]
Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/coins.ts]: Done

File: modules/statics/src/coins.ts [hunk 18]
```diff
@@ -1117,4 +1121,44 @@ export const coins = CoinMap.fromCoins([
...
+  avaxErc20('QI', 'BenQi', 18, '0x8729438eb15e2c8b576fcc6aecda6a148776c0f5', UnderlyingAsset.QI),
+  avaxErc20('XAVA', 'Avalaunch', 18, '0xd1c3f94de7e5b45fa4edbba472491a9f4b166fc4', UnderlyingAsset.XAVA),
+  avaxErc20('KLO', 'Kalao', 18, '0xb27c8941a7df8958a1778c0259f76d1f8b711c35', UnderlyingAsset.KLO),
+  avaxErc20('JOE', 'Trader Joe', 18, '0x6e84a6216ea6dacc71ee8e6b0a5b7322eebc0fdd', UnderlyingAsset.JOE),
+  arc20Token(
+    'PNG',
+    'Pangolin',
+    18,
+    '0x60781c2586d68229fde47564546784ab3faca982',
+    '2ugMhQ5ygoZ721Hjka413YkzNVGMjKQzmaDkxpeimKYEnAaWiD',
```
[the remark this answers is not in the document]
Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/coins.ts]: Done

File: modules/statics/src/base.ts [hunk 19]
```diff
@@ -719,6 +719,13 @@ export enum UnderlyingAsset {
   CHEX = 'chex',
   IQ = 'iq',
   EOS_BOX = 'eos:box',
+
+  // Avax Token ERC-20
+  QI = 'qi',
+  XAVA = 'xava',
+  KLO = 'klo',
+  JOE = 'joe',
+  PNG = 'png',
```
Comment (shiksb) [inline modules/statics/src/base.ts:728]: after some discussion with @CapnMigraine, we should add them as follows:

```suggestion
avaxc:QI = 'avaxc:qi',
```

Comment (charly-simtlix) [reply modules/statics/src/base.ts:728]: Question, the other tokens were removed or should be modified the same way than "qi" ?

Comment (tylerlevine) [reply modules/statics/src/base.ts:728]: If the asset exists on two chains, then the tokens which represent it on each chain should share the same underlying asset

Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/base.ts]: Done

**Review by tylerlevine — APPROVED** [review]

File: modules/statics/src/account.ts (hunk 1 above)
Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/account.ts]: Done!

File: modules/statics/src/base.ts (hunk 16 above)
Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/base.ts]: Done!

**Comment by CapnMigraine** [comment]
@shiksb do we still need this?

**Review by sskim16 — APPROVED** [review]

**Review by sskim16 — APPROVED** [review]

File: modules/statics/src/coins.ts [hunk 20]
```diff
@@ -1120,4 +1124,22 @@ export const coins = CoinMap.fromCoins([
...
+  avaxErc20('PNG', 'Pangolin', 18, '0x60781c2586d68229fde47564546784ab3faca982', UnderlyingAsset['avaxc:PNG']),
+  avaxErc20('USDT', 'Tether', 6, '0xc7198437980c041c805a1edcba50c1ce5db95118', UnderlyingAsset.USDT),
+  tavaxErc20('tQI', 'Test BenQi', 18, '0x8729438eb15e2c8b576fcc6aecda6a148776c0f5', UnderlyingAsset['avaxc:QI']),
+  tavaxErc20(
+    'tXAVA',
+    'Test Avalaunch',
+    18,
+    '0xd1c3f94de7e5b45fa4edbba472491a9f4b166fc4',
+    UnderlyingAsset['avaxc:XAVA']
+  ),
```
Comment (shiksb) [inline modules/statics/src/coins.ts:1140]: nit: formatting

File: modules/statics/src/coins.ts [hunk 21]
```diff
@@ -1120,4 +1124,22 @@ export const coins = CoinMap.fromCoins([
...
+  tavaxErc20('tKLO', 'Test Kalao', 18, '0xb27c8941a7df8958a1778c0259f76d1f8b711c35', UnderlyingAsset['avaxc:KLO']),
+  tavaxErc20('tJOE', 'Test Trader Joe', 18, '0x6e84a6216ea6dacc71ee8e6b0a5b7322eebc0fdd', UnderlyingAsset['avaxc:JOE']),
+  tavaxErc20('tPNG', 'Test Pangolin', 18, '0x60781c2586d68229fde47564546784ab3faca982', UnderlyingAsset['avaxc:PNG']),
+  tavaxErc20('tUSDT', 'Test Tether', 6, '0xc7198437980c041c805a1edcba50c1ce5db95118', UnderlyingAsset.USDT),
```
Comment (shiksb) [inline modules/statics/src/coins.ts:1144]: should USDC be here?

File: modules/statics/src/coins.ts [hunk 22]
```diff
@@ -1120,4 +1124,22 @@ export const coins = CoinMap.fromCoins([
   teosToken('teos:IQ', 'Everipedia', 3, 'testtoken112', UnderlyingAsset.IQ, AccountCoin.DEFAULT_FEATURES, '', 'IQ'),
   teosToken('teos:BOX', 'Box', 6, 'kvszn1xyz1bu', UnderlyingAsset.EOS_BOX, AccountCoin.DEFAULT_FEATURES, '', 'BOX'),
   teosToken('teos:USDT', 'Tether', 4, 'lionteste212', UnderlyingAsset.USDT, AccountCoin.DEFAULT_FEATURES, '', 'USDT'),
+  avaxErc20('QI', 'BenQi', 18, '0x8729438eb15e2c8b576fcc6aecda6a148776c0f5', UnderlyingAsset['avaxc:QI']),
+  avaxErc20('XAVA', 'Avalaunch', 18, '0xd1c3f94de7e5b45fa4edbba472491a9f4b166fc4', UnderlyingAsset['avaxc:XAVA']),
+  avaxErc20('KLO', 'Kalao', 18, '0xb27c8941a7df8958a1778c0259f76d1f8b711c35', UnderlyingAsset['avaxc:KLO']),
+  avaxErc20('JOE', 'Trader Joe', 18, '0x6e84a6216ea6dacc71ee8e6b0a5b7322eebc0fdd', UnderlyingAsset['avaxc:JOE']),
+  avaxErc20('PNG', 'Pangolin', 18, '0x60781c2586d68229fde47564546784ab3faca982', UnderlyingAsset['avaxc:PNG']),
+  avaxErc20('USDT', 'Tether', 6, '0xc7198437980c041c805a1edcba50c1ce5db95118', UnderlyingAsset.USDT),
```
Comment (shiksb) [inline modules/statics/src/coins.ts:1132]: why not 
```suggestion
  avaxErc20('USDT', 'Tether', 6, '0xc7198437980c041c805a1edcba50c1ce5db95118', UnderlyingAsset['avaxc:USDT']),
```

File: modules/statics/src/base.ts [hunk 23]
```diff
@@ -719,6 +719,13 @@ export enum UnderlyingAsset {
...
   IQ = 'iq',
   EOS_BOX = 'eos:box',
+
+  // Avax Token ERC-20
+  'avaxc:QI' = 'avaxc:qi',
+  'avaxc:XAVA' = 'avaxc:xava',
+  'avaxc:KLO' = 'avaxc:klo',
+  'avaxc:JOE' = 'avaxc:joe',
+  'avaxc:PNG' = 'avaxc:png',
 }
```
Comment (shiksb) [inline modules/statics/src/base.ts:729]: where is USDT?

**Review by shiksb — CHANGES_REQUESTED** [review]

File: modules/statics/src/account.ts [hunk 24]
```diff
@@ -888,3 +916,175 @@ export function tsolToken(
...
+    contractAddress,
+    assetIdXChain,
+    asset,
+    features,
+    prefix,
+    suffix,
+    network,
+    primaryKeyCurve
+  );
+}
```
Comment (sskim16) [inline modules/statics/src/account.ts:1090]: No longer need ARC-20 support

File: modules/statics/src/account.ts [hunk 25]
```diff
@@ -888,3 +916,175 @@ export function tsolToken(
...
+      prefix,
+      suffix,
+      features,
+      decimalPlaces,
+      asset,
+      isToken: true,
+      primaryKeyCurve,
+    })
+  );
+}
```
Comment (sskim16) [inline modules/statics/src/account.ts:1047]: ARAC-20 support no longer needed

**Review by sskim16 — CHANGES_REQUESTED** [review]

File: modules/statics/src/account.ts [hunk 26]
```diff
@@ -254,6 +258,30 @@ export class SolCoin extends AccountCoinToken {
...
+ */
+export class ARC20Token extends AvaxERC20Token {
+  public assetIdXChain: string;
+  constructor(options: ARC20CoinContructorOptions) {
+    super(options);
+
+    this.assetIdXChain = options.assetIdXChain;
+  }
+}
+
```
Comment (sskim16) [inline modules/statics/src/account.ts:284]: ARC-20 no longer needed

File: modules/statics/src/account.ts [hunk 27]
```diff
@@ -74,6 +74,10 @@ export interface SolCoinConstructorOptions extends AccountConstructorOptions {
   tokenAddress: string;
 }
 
+export interface ARC20CoinContructorOptions extends Erc20ConstructorOptions {
+  assetIdXChain: string;
+}
+
```
Comment (sskim16) [inline modules/statics/src/account.ts:80]: arc-20 no longer needed

**Review by sskim16 — DISMISSED** [review]

File: modules/statics/src/coins.ts (hunk 21 above)
Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/coins.ts:1144]: there isn't information in the PRD about usdc

**Review by jrafael-simtlix — DISMISSED** [review]

File: modules/statics/src/base.ts [hunk 28]
```diff
@@ -720,6 +720,15 @@ export enum UnderlyingAsset {
...
   IQ = 'iq',
   EOS_BOX = 'eos:box',
+
+  // Avax Token ERC-20
+  'avaxc:QI' = 'avaxc:qi',
+  'avaxc:XAVA' = 'avaxc:xava',
+  'avaxc:KLO' = 'avaxc:klo',
+  'avaxc:JOE' = 'avaxc:joe',
+  'avaxc:PNG' = 'avaxc:png',
+  'avaxc:USDT' = 'avaxc:usdt',
```
Comment (jicli) [inline modules/statics/src/base.ts:730]: should this be usdt.e?

Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/base.ts:730]: Yes, and too should this be usdc.e???

Comment (mauricio-ferreyra) [reply modules/statics/src/base.ts:730]: Yes, is the answer to both questions.

Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/base.ts]: Done!

**Review by charly-simtlix — APPROVED** [review]

File: modules/statics/src/coins.ts (hunk 20 above)
Comment (shiksb) [reply modules/statics/src/coins.ts:1140]: okay

File: modules/statics/src/coins.ts [hunk 29]
```diff
@@ -1121,4 +1123,36 @@ export const coins = CoinMap.fromCoins([
   teosToken('teos:IQ', 'Everipedia', 3, 'testtoken112', UnderlyingAsset.IQ, AccountCoin.DEFAULT_FEATURES, '', 'IQ'),
   teosToken('teos:BOX', 'Box', 6, 'kvszn1xyz1bu', UnderlyingAsset.EOS_BOX, AccountCoin.DEFAULT_FEATURES, '', 'BOX'),
   teosToken('teos:USDT', 'Tether', 4, 'lionteste212', UnderlyingAsset.USDT, AccountCoin.DEFAULT_FEATURES, '', 'USDT'),
+  avaxErc20('QI', 'BenQi', 18, '0x8729438eb15e2c8b576fcc6aecda6a148776c0f5', UnderlyingAsset['avaxc:QI']),
```
Comment (shiksb) [inline modules/statics/src/coins.ts:1126]: @jrafael-simtlix should the first parameter be the underlying asset name for all tokens? For example should it be this? 

```suggestion
  avaxErc20('avaxc:QI', 'BenQi', 18, '0x8729438eb15e2c8b576fcc6aecda6a148776c0f5', UnderlyingAsset['avaxc:QI']),
```

File: modules/statics/src/coins.ts [hunk 30]
```diff
@@ -1121,4 +1123,36 @@ export const coins = CoinMap.fromCoins([
...
+  avaxErc20('XAVA', 'Avalaunch', 18, '0xd1c3f94de7e5b45fa4edbba472491a9f4b166fc4', UnderlyingAsset['avaxc:XAVA']),
+  avaxErc20('KLO', 'Kalao', 18, '0xb27c8941a7df8958a1778c0259f76d1f8b711c35', UnderlyingAsset['avaxc:KLO']),
+  avaxErc20('JOE', 'Trader Joe', 18, '0x6e84a6216ea6dacc71ee8e6b0a5b7322eebc0fdd', UnderlyingAsset['avaxc:JOE']),
+  avaxErc20('PNG', 'Pangolin', 18, '0x60781c2586d68229fde47564546784ab3faca982', UnderlyingAsset['avaxc:PNG']),
+  avaxErc20('USDT', 'Tether USD', 6, '0xc7198437980c041c805a1edcba50c1ce5db95118', UnderlyingAsset['avaxc:USDT.E']),
+  avaxErc20('USDC', 'USD Coin', 6, '0xa7d7079b0fead91f3e65f86e8915cb59c1a4c664', UnderlyingAsset['avaxc:USDC.E']),
+  tavaxErc20('tQI', 'Test BenQi', 18, '0x8729438eb15e2c8b576fcc6aecda6a148776c0f5', UnderlyingAsset['avaxc:QI']),
```
Comment (shiksb) [inline modules/statics/src/coins.ts:1133]: should this be 
```suggestion
  tavaxErc20('tavaxc:QI', 'Test BenQi', 18, '0x8729438eb15e2c8b576fcc6aecda6a148776c0f5', UnderlyingAsset['avaxc:QI']),
```

File: modules/statics/src/coins.ts (hunk 29 above)
Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/coins.ts]: Done

File: modules/statics/src/coins.ts (hunk 30 above)
Comment (JoaquinTissera-Simtlix) [reply modules/statics/src/coins.ts]: Done

**Review by jrafael-simtlix — APPROVED** [review]

**Review by shiksb — APPROVED** [review]

**Pull request merged**